### PR TITLE
Add port ranges

### PR DESCRIPTION
**Pull request by baude**
Use the ability of ranges with iptables instead of creating multiple
rules for each port.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: test/100-bridge-iptables.bats [hunk 1]
```diff
@@ -12,24 +12,6 @@ fw_driver=iptables
     assert "${lines[0]}" "==" "[INFO  netavark::firewall] Using iptables firewall driver" "iptables driver is in use"
 }
 
-@test "$fw_driver - internal network" {
```
Comment (Luap99) [inline test/100-bridge-iptables.bats:15]: Why is this removed?

File: test/100-bridge-iptables.bats [hunk 2]
```diff
@@ -145,3 +127,19 @@ fw_driver=iptables
     setup_sctp_kernel_module
     test_port_fw ip=dual proto=sctp
 }
+
+@test "$fw_driver - port range forwarding ipv4 - tcp" {
+    test_port_fwrange=3
```
Comment (Luap99) [inline test/100-bridge-iptables.bats:132]:
```suggestion
    test_port_fw range=3
```

File: test/100-bridge-iptables.bats [hunk 3]
```diff
@@ -145,3 +127,19 @@ fw_driver=iptables
...
+}
+
+@test "$fw_driver - port range forwarding ipv6 - tcp" {
+    test_port_fw ip=6 range=3
+}
```
Comment (Luap99) [inline test/100-bridge-iptables.bats:137]: Can you also add a test with ip=dual

**Review by Luap99 — COMMENTED** [review]
The tests are failing, it looks like always the second ncat fails with connection refused.

File: test/100-bridge-iptables.bats (hunk 2 above)
Comment (flouthoc) [reply test/100-bridge-iptables.bats:132]: @baude This is not yet fixed

**Comment by flouthoc** [comment]
@baude I think there is something up with the code. Cause the example you were trying with `-p 8080-8082:80-82` works with `main` branch without this PR. So podman is for sure splitting port forward into seperate entries rather than using `range` field.

I wonder if we only need this PR for `-p 8080-8082:80-82` then it works with `firewall` disabled from the current `main` without any addition.

For `range` part i think there is something up with the code cause tests looked fine to me.

**Comment by Luap99** [comment]
@flouthoc You can run with `--log-level trace`, this will show you the exact config for netavark. I am pretty sure `-p 8080-8082:80-82` creates a range, if not than this is a bug

**Comment by flouthoc** [comment]
@Luap99 I'll try that but could you also confirm if this works for you from `main` branch. `run -it -p 8080:8082:80-82 quay.io/libpod/alpine_nginx` and make sure you switch the backend to `netavark`.

**Comment by Luap99** [comment]
I am on PTO. I have no access to my workstation so I cannot test netavark

**Review by flouthoc — CHANGES_REQUESTED** [review]
@baude There is a tiny logical error or typo in `DNAT` rule `--dport` must accept `host-range`. Following diff should make it work and CI should also pass.

Could you please amend PR with following diff.

```diff
diff --git a/src/firewall/varktables/types.rs b/src/firewall/varktables/types.rs
index 05df4e1..c6884ae 100644
--- a/src/firewall/varktables/types.rs
+++ b/src/firewall/varktables/types.rs
@@ -418,7 +418,7 @@ pub fn get_port_forwarding_chains<'a>(
                 i.protocol,
                 container_ip_value,
                 container_port,
-                i.host_port.to_string(),
+                &host_port
             ),
             None,
         ));
```

**Comment by flouthoc** [comment]
@baude cargo fmt is needed.

**Review by flouthoc — APPROVED** [review]

**Comment by Luap99** [comment]
Please restore the internal test and add the ip=dual case as noted in the review e ts

**Pull request merged**